src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -251,12 +267,18 @@ data Level m h = Level {
 
 -- | A merging run is either a single run, or some ongoing merge.
 data MergingRun m h =
-    MergingRun !(StrictMVar m (MergingRunState m h))
+    MergingRun !MergePolicyForLevel !NumRuns !(StrictMVar m (MergingRunState m h))
   | SingleRun !(Run m h)
 
+newtype NumRuns = NumRuns { unNumRuns :: Int }
+  deriving stock (Show, Eq)
+
 data MergingRunState m h =
     CompletedMerge !(Run m h)
-  | OngoingMerge !(V.Vector (Run m h)) !(Merge m h)
+  | OngoingMerge !(V.Vector (Run m h)) !(PrimVar (PrimState m) Int {- NumStepsDone -}) !(Merge m h)
+
+newtype NumStepsDone = NumStepsDone { unNumStepsDone :: Int }
```

Comment:
I agree this variable should be documented. In our code we really only use "steps" as a term for "how many steps did my merge progress", so I thought it would at a glance be clear that that is what it is counting

As mentioned before, I can add more sanity checks

Reply:
I've created #445 so that we can sanity check the implementation against the prototype later.

The accounting of `NumStepsDone` has also been reworked slightly, and the rework should have more documentation now